src/PostgREST/Middleware.hs

```diff
@@ -50,9 +51,16 @@ unsupportedAccept app req respond =
         isTargetRoot = fromMaybe True $ (== "") <$> listToMaybe path
         unsupportedAcceptRespond = respond $ errResponse status415 "Unsupported Accept header, try: application/json"
 
+malformedProxyUri :: Application -> Application
+malformedProxyUri app req respond =
+  if isMalformedProxyUri proxyUri then malformedProxyUriRespond else app req respond
+  where proxyUri = lookup "x-forwarded-uri" $ requestHeaders req
+        malformedProxyUriRespond = respond $ errResponse status400 $ "Malformed X-Forwarded-URI header: " <> cs (fromJust proxyUri)
+
 defaultMiddle :: Application -> Application
 defaultMiddle =
     gzip def
   . cors corsPolicy
   . staticPolicy (only [("favicon.ico", "static/favicon.ico")])
   . unsupportedAccept
+  . malformedProxyUri
```

Comment:
while behind the proxy, there is going to be a proxy uri header on each request, and it's going to get checked and parsed on each request just to confirm it's correct even though you do nothing with it later.
there could be an argument for using this in the location header but it seems too much computation with not much benefit. the location header can be easily adjusted by the proxy with
`add_header Content-Location  /prefix/goes/here/$upstream_http_content_location;`